File: problems/prob13/prob13.js
/*

Write a function that takes two random integers as arguments and returns the least common multiple.

*/

function lcm(num1, num2) {
  return (num1 * num2) / gcd(num1, num2);
}

function gcd(num1, num2) {
  return !num2 ? num1 : gcd(num2, num1 % num2);
}

const num1 = Math.floor((Math.random() * 100));
const num2 = Math.floor((Math.random() * 100));
console.log(`The numbers are ${num1} and ${num2}`);
console.log(lcm(num1, num2));

File: problems/prob13/prob13.py
from random import randrange


def lcm(num1, num2):
    return num1 * num2 // gcd(num1, num2)


def gcd(num1, num2):
    while num2:
        num1, num2 = num2, num1 % num2
    return num1

rand1 = randrange(1, 100)
rand2 = randrange(1, 100)
print("The numbers are " + str(rand1) + " and " + str(rand2))
print(lcm(rand1, rand2))
